src/api/middlewares/translation_middleware.py

```diff
@@ -29,13 +32,15 @@ def __call__(self, request):
 			original_content = response.content.decode('utf-8')
 			response_to_dict = json.loads(original_content)
 			
-			preferred_language = request.POST.get('__preferred_language', "en-US")
-			target_destination_language = request.POST.get('__user_language', preferred_language)
+			preferred_language = request.POST.get('__preferred_language', DEFAULT_SOURCE_LANGUAGE_CODE)
+			user_language = request.POST.get('__user_language', preferred_language)
+			
+			supported_language_target_language = get_supported_language(user_language)
 			
-			if target_destination_language == 'en-US':  #TODO remove this when we start supporting data upload in other languages
+			if supported_language_target_language == DEFAULT_SOURCE_LANGUAGE_CODE:  #TODO remove this when we start supporting data upload in other languages
 				return response
 				
-			target_language_code = to_third_party_lang_code(target_destination_language)
+			target_language_code = to_third_party_lang_code(supported_language_target_language)
```

Comment:
Nit picky but can you change `supported_language_target_language` to `target_language`?